### PR TITLE
GH-970: Add `AmqpAppender.verifyHostname` option

### PR DESCRIPTION
**Pull request by artembilan**
Fixes https://github.com/spring-projects/spring-amqp/issues/970

* Some code polishing and internal refactoring for `AmqpAppender`

**Cherry-pick to 2.1.x**

### CODE REVIEW

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/logback/AmqpAppender.java [hunk 1]
```diff
@@ -620,7 +641,7 @@ public void start() {
 		if (rabbitConnectionFactory != null) {
 			super.start();
 			this.routingKeyLayout.setPattern(this.routingKeyLayout.getPattern()
-					.replaceAll("%property\\{applicationId\\}", this.applicationId));
+					.replaceAll("%property\\{applicationId}", this.applicationId));
```
Comment (garyrussell) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/logback/AmqpAppender.java]: This looks wrong, curly brackets must be escaped in RegEx.

Isn't there a test for this?

Comment (artembilan) [reply spring-rabbit/src/main/java/org/springframework/amqp/rabbit/logback/AmqpAppender.java]: This is what my IDEA shows me:

изображение

And yes: it is covered by our tests.
Look into `logback-test.xml` and `log4j2-amqp-appender.xml` and search for `{applicationId}`

**Comment by garyrussell** [comment]
I failed to cherry-pick this; 462efc43fcf46d4d80d30fd0a07dbcace06dd907

**Comment by artembilan** [comment]
OK. I'll do that myself later or issue a back-port PR.
Thanks

**Comment by artembilan** [comment]
Wait. Looks like you have figured that out: https://github.com/spring-projects/spring-amqp/commit/462efc43fcf46d4d80d30fd0a07dbcace06dd907

Thanks

**Comment by garyrussell** [comment]
Sorry, yes; the link to the cherry pick was in my comment; I should have made that clear.

**Pull request merged**